src/platform/forms-system/src/js/components/SchemaForm.jsx

```diff
@@ -101,8 +103,12 @@ class SchemaForm extends React.Component {
 
   onBlur(id) {
     if (!this.state.formContext.touched[id]) {
-      const formContext = set(['touched', id], true, this.state.formContext);
-      this.setState({ formContext });
+      const data = getFormDataFromSchemaId(id, this.props.data);
+      const isEmpty = data === undefined || data === null || data === '';
+      if (!isEmpty) {
+        const formContext = set(['touched', id], true, this.state.formContext);
+        this.setState({ formContext });
+      }
```

Comment:
I _feel_ like this won't ever set a field as touched if the data is empty - like what happens if I focus, type in a bunch of stuff, then clear and blur - no error would appear, right?

Should we include a flag that some interaction occurred within the form element (besides tabbing) and then set the `touched` flag?

Reply:
if they leave the field and there's nothing in there, that's basically considered not an attempt (only on the first try). once they finally put something in there and leave the field, from that point forward, they will always see errors, including if they try to empty it again. 

however if their data was prefilled, and they go to clear it, that's an edge case where doesn't act perfectly - because they won't see an error. however they will see the error when they try to continue to the next page, so it's not such a big deal.

adding an additional flag to check for state changes would add a lot more complexity for a little bit of benefit, so I think this current solution is a reasonable compromise.